spring-integration-core/src/main/java/org/springframework/integration/mapping/AbstractHeaderMapper.java

```diff
@@ -402,10 +429,17 @@ private boolean containsIgnoreCase(String name) {
 
 		private final Collection<String> patterns;
 
+		private final boolean negate;
+
 		public PatternBasedHeaderMatcher(Collection<String> patterns) {
+			this(patterns, false);
```

Comment:
It is outside the PR code, but I would like to share concern:
```
for (String pattern : this.patterns) {
				if (PatternMatchUtils.simpleMatch(pattern.toLowerCase(), header)) {
```
The `pattern.toLowerCase()` should be done only once during initialization, but not on each header.